msprime/species_tree.py

```diff
@@ -0,0 +1,834 @@
+
+    def get_id(self):
+        return self.id
+
+    def set_node_ids(self, node_ids):
+        self.node_ids = node_ids
+
+    def get_node_ids(self):
+        return self.node_ids
+
+    def get_is_extant(self):
+        if self.node_ids[1][0:12] == 'internalNode':
+            return False
+        else:
+            return True
+
+    def set_length(self, length):
+        self.length = length
+
+    def get_length(self):
+        return self.length
+
+    def set_is_root_edge(self, is_root_edge):
+        self.is_root_edge = is_root_edge
+
+    def get_is_root_edge(self):
+        return self.is_root_edge
+
+    def add_extant_progeny_id(self, extant_progeny_id):
+        self.extant_progeny_ids.append(extant_progeny_id)
+
+    def get_extant_progeny_ids(self):
+        return self.extant_progeny_ids
+
+    def set_termination(self, termination):
+        self.termination = termination
+
+    def get_termination(self):
+        return self.termination
+
+    def set_origin(self, origin):
+        self.origin = origin
+
+    def get_origin(self):
+        return self.origin
+
+    def set_parent_needs_times(self, parent_needs_times):
+        self.parent_needs_times = parent_needs_times
+
+    def get_parent_needs_times(self):
+        return self.parent_needs_times
+
+    def set_dmv(self, dmv):
+        self.dmv = dmv
+
+    def get_pop_size(self, generations_per_branch_length_unit):
+        return self.dmv * generations_per_branch_length_unit
+
+    def info(self):
+        info_string = ''
+        info_string += 'Edge id:'.ljust(28)
+        info_string += self.id
+        info_string += '\n'
+        info_string += 'Edge node 1 id:'.ljust(28)
+        info_string += self.node_ids[0]
+        info_string += '\n'
+        info_string += 'Edge node 2 id:'.ljust(28)
+        info_string += self.node_ids[1]
+        info_string += '\n'
+        info_string += 'Edge length:'.ljust(28)
+        info_string += str(self.length)
+        info_string += '\n'
+        if self.dmv is not None:
+            info_string += 'Edge dmv:'.ljust(28)
+            info_string += str(self.dmv)
+            info_string += '\n'
+        info_string += 'Edge origin:'.ljust(28)
+        info_string += str(self.origin)
+        info_string += '\n'
+        info_string += 'Edge termination:'.ljust(28)
+        info_string += str(self.termination)
+        info_string += '\n'
+        info_string += 'Edge is extant:'.ljust(28)
+        info_string += str(self.get_is_extant())
+        info_string += '\n'
+        info_string += 'Edge is root edge:'.ljust(28)
+        info_string += str(self.is_root_edge)
+        info_string += '\n'
+        info_string += 'Edge extant progeny ids:'.ljust(28)
+        for extant_progeny_id in self.extant_progeny_ids:
+            info_string += '{}, '.format(extant_progeny_id)
+        info_string = info_string[:-2]
+        info_string += '\n'
+        return info_string
+
+
+def parse_species_tree(
+        filename=None,
```

Comment:
I think it would be simpler for now to pass a string as the argument rather than insisting on a file path. It makes things pretty tedious for testing if we have to write a bunch of tempfiles. It's easy enough for the user to write
```python
with open("treefile", "r") as f:
     parse_species_tree(f.read())
```

We can add an additional ``filename`` argument also later if you like, but for the initial version let's get it working with a string parameter.

Reply:
I agree, this is now changed.